data/src/main/java/com/example/pinterbest/data/repository/PinsRepositoryImpl.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.pinterbest.data.repository
+
+import android.content.SharedPreferences
+import com.example.pinterbest.data.api.ApiClient
+import com.example.pinterbest.data.models.toPin
+import com.example.pinterbest.domain.common.ErrorMessage
+import com.example.pinterbest.domain.common.Result
+import com.example.pinterbest.domain.entities.Pin
+import com.example.pinterbest.domain.entities.PinsList
+import com.example.pinterbest.domain.repositories.PinsRepository
+import java.io.IOException
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import retrofit2.HttpException
+
+class PinsRepositoryImpl(val preferences: SharedPreferences) : PinsRepository {
+    override suspend fun getPins(): Flow<Result<PinsList>> = flow {
+        try {
+            emit(Result.Loading)
+            val pins = ApiClient().getInstance().getClient().getPinFeed(
+                SessionRepositoryImpl(preferences).authProvider() ?: ""
```

Comment:
SessionRepositoryImpl must be provided outside of class via constructor. Creating any dependencies inside it's a bad practice

Reply:
Fixed